sources/core/Stride.Core.DiagnosticsServices/ClassInfo.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace StrideDiagnostics;
+public class ClassInfo
+{
+    public GeneratorExecutionContext ExecutionContext { get; set; }
+    public TypeDeclarationSyntax TypeSyntax { get; set; }
+    public ClassDeclarationSyntax SerializerSyntax { get; set; }
+    public TypeParameterListSyntax Generics => TypeSyntax.TypeParameterList;
+    public bool IsGeneric => TypeSyntax.TypeParameterList != null && TypeSyntax.TypeParameterList.Parameters.Count > 0;
+    public INamedTypeSymbol Symbol { get; set; }
+    public string TypeName { get; set; }
+    public string SerializerName { get; set; }
+    public string NamespaceName { get; set; }
+    public StrideDiagnosticsSyntaxReceiver SyntaxReceiver { get; set; }
+    public SemanticModel SemanticModel { get => _compilationCache ??= ExecutionContext.Compilation.GetSemanticModel(TypeSyntax.SyntaxTree); }
+    private SemanticModel _compilationCache;
+    private List<IPropertySymbol> _propertyCache;
+    public IEnumerable<IPropertySymbol> AvailableProperties
+    {
+        get
+        {
+            if (_propertyCache == null)
+            {
+                INamedTypeSymbol classSymbol = SemanticModel.GetDeclaredSymbol(TypeSyntax);
+                IEnumerable<IPropertySymbol> properties = PropertyAttributeFinder.FilterBasePropertiesRecursive(ref classSymbol);
+                return _propertyCache = properties.ToList();
+            }
+            else
+            {
+                return _propertyCache;
+            }
+        }
+    }
+    public bool IsAbstract()
```

Comment:
Should this be a property as well, like `IsGeneric`?

Reply:
[29d7e7d](https://github.com/stride3d/stride/pull/1778/commits/29d7e7d2b76ce6c56cf0063307a2aee8b10bf015)

is now a property